tests/unit/core/test_utils.js

```diff
@@ -957,3 +972,107 @@ describe('getClientCompatibility', () => {
     });
   });
 });
+
+describe('getCategoryColor', () => {
+  it('handles a false-y category', () => {
+    const category = null;
+    const categoryColor = getCategoryColor(category);
+
+    expect(categoryColor).toEqual(1);
+  });
+
+  it('all valid addonTypes are in CATEGORY_COLORS', () => {
+    expect(() => {
+      validAddonTypes.forEach((addonType) => {
+        const category = { id: 2, type: addonType };
+
+        getCategoryColor(category);
+      });
+    }).not.toThrowError('not found in CATEGORY_COLORS');
+  });
+
+  it('throws on unrecognised addonType', () => {
+    expect(() => {
+      getCategoryColor({ id: 5, type: 'NOT_A_REAL_TYPE' });
+    }).toThrowError(
+      'addonType "NOT_A_REAL_TYPE" not found in CATEGORY_COLORS');
+  });
+
+  it('deals with hundreds of categories', () => {
+    for (let i = 1; i < 800; i++) {
```

Comment:
Is it really necessary to run this 800 times? It seems a bit excessive and might slow down the test suite. Can you do a smoke test of the upper range instead somehow?